internal/facade/client.go

```diff
@@ -80,14 +91,155 @@ type SpaceClient interface {
+		instances:    make(map[string]*Instance),
+		bindings:     make(map[string]*Binding),
+		cacheTimeOut: 5 * time.Minute,
+	}
+	return cache
+}
+
+func (c *ResourceCache) GetCachedInstances() map[string]*Instance {
+	return c.instances
+}
+
+func (c *ResourceCache) GetCachedBindings() map[string]*Binding {
+	return c.bindings
+}
+
+func (c *ResourceCache) GetCachedSpaces() map[string]*Space {
+	return c.spaces
+}
+
+// function to set the resource cache enabled flag from config
+func (c *ResourceCache) SetResourceCacheEnabled(enabled bool) {
+	c.isResourceCacheEnabled = enabled
+}
+func (c *ResourceCache) IsResourceCacheEnabled() bool {
+	if c == nil {
+		return false
+	}
+	return c.isResourceCacheEnabled
+}
+
+// Function to set the resource cache enabled flag from config
+func (c *ResourceCache) GetCacheTimeOut() time.Duration {
+	return c.cacheTimeOut
+}
+
+func (c *ResourceCache) SetCacheTimeOut(timeOut string) {
+	cacheTimeOut, err := time.ParseDuration(timeOut)
+	if err != nil {
+		fmt.Printf("Error parsing duration: %v\n", err)
+		return
+	}
+	c.cacheTimeOut = cacheTimeOut
+}
+
+// Function to check if the cache is expired
+func (c *ResourceCache) IsCacheExpired() bool {
+
+	expirationTime := c.lastCacheTime.Add(c.cacheTimeOut)
+	//expiryTime := time.Until(c.lastCacheTime)
+	fmt.Printf("Expiry time: %v\n", expirationTime)
+	fmt.Printf("Cache timeout: %v\n", c.cacheTimeOut)
+	return time.Now().After(expirationTime)
+
+}
+
+// Function to set the last cache time
+func (c *ResourceCache) SetLastCacheTime() {
+	c.lastCacheTime = time.Now()
+	fmt.Printf("Last cache time: %v\n", c.lastCacheTime)
+}
+
+// AddSpaceInCache stores a space in the cache
+func (c *ResourceCache) AddSpaceInCache(key string, space *Space) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	c.spaces[key] = space
+}
+
+// GetSpaceFromCache retrieves a space from the cache
+func (c *ResourceCache) GetSpaceFromCache(key string) (*Space, bool) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	space, found := c.spaces[key]
+	return space, found
 }
 
-type SpaceClientBuilder func(string, string, string, string) (SpaceClient, error)
+// AddInstanceInCache stores an instance in the cache
+func (c *ResourceCache) AddInstanceInCache(key string, instance *Instance) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	c.instances[key] = instance
+}
+
+// GetInstanceFromCache retrieves an instance from the cache
+func (c *ResourceCache) GetInstanceFromCache(key string) (*Instance, bool) {
+	c.mutex.RLock()
+	defer c.mutex.RUnlock()
+	instance, found := c.instances[key]
+	return instance, found
+}
+
+// RemoveInstanceFromCache removes an instance from the cache
+// This is used when an instance is deleted
+// The instance is removed from the cache to avoid stale data
+// The instance is removed from the cache only if the instance is found in the cache
+func (c *ResourceCache) RemoveInstanceFromCache(key string) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	_, found := c.instances[key]
```

Comment:
just calling "delete(c.instances, key)" directly should also work

Reply:
Adopted the code accordingly